### PR TITLE
[MM-18589] Fix channel mention autocomplete behaviour

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes a couple of problems with the channel mention autocomplete provider:

*Other Channels* list wasn't populated and the final channel list was not properly kept in sync.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-18589

### CODE REVIEW

File: components/suggestion/channel_mention_provider.jsx [hunk 1]
```diff
@@ -3,15 +3,13 @@
...
+import {getMyChannels, getMyChannelMemberships} from 'mattermost-redux/selectors/entities/channels';
 
 import {sortChannelsByTypeAndDisplayName} from 'mattermost-redux/utils/channel_utils';
 
-import {ChannelTypes} from 'mattermost-redux/action_types';
-
 import store from 'stores/redux_store.jsx';
 
-import {Constants} from 'utils/constants';
+import {Constants} from 'utils/constants.jsx';
```
Comment (jespino) [inline components/suggestion/channel_mention_provider.jsx]: Why this change is needed?

Comment (streamer45) [reply components/suggestion/channel_mention_provider.jsx]: I think that was automated linting.

Comment (hmhealey) [reply components/suggestion/channel_mention_provider.jsx]: 0/5 on this change now, but it's going to get reverted when that file is switched to TypeScript. You'll probably need to figure out how to disable that in the future

Comment (streamer45) [reply components/suggestion/channel_mention_provider.jsx]: That was `npm run fix` by the way.

Comment (jespino) [reply components/suggestion/channel_mention_provider.jsx]: Anyway, I would revert this change because is going to be typescript soon (I think).

Comment (streamer45) [reply components/suggestion/channel_mention_provider.jsx]: No problem :+1:

**Comment by streamer45** [comment]
> Is this something we could add tests for?

I think it might be tricky to properly test this. Maybe e2e could be an option. Will look into it.
